File: problems/0001_two-sum/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val m = HashMap<Int, Int>(nums.size + 1, 0.99f)
        nums.forEachIndexed { i, num ->
            val j = m[target - num]
            if (j != null)
                return intArrayOf(j, i)
            m.put(num, i)
        }
        throw Exception("No match value for $nums, $target!")
    }
}

File: problems/0001_two-sum/Solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var m = new Dictionary<int, int>(nums.Length + 1);
        for (int i = 0; i < nums.Length; ++i) {
            var num = nums[i];
            var r = target - num;
            if (m.ContainsKey(r))
                return new int[] { m[r], i };
            m[num] = i;
        }
        throw new System.Exception($"No match value for {nums}, {target}!");
    }
}
